# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1220455225 14400
#      Wed Sep 03 11:20:25 2008 -0400
# Node ID 163d20e4d39c450943e56e3b366ce671e384c232
# Parent  b1634dd9ebe7a21c78ed9cf2341d81a9efbef569
symtab.cc (out_of_date_check_internal): check for method, then regular function

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-09-03  John W. Eaton  <jwe@octave.org>
+
+	* symtab.cc (out_of_date_check_internal): If dispatch type is set,
+	check for method.  If that fails, check for regular function. 
+
 2008-09-02  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* graphics.cc (hggroup::update_axis_limits): Also reacts on
 	[xyzac]liminclude changes.
 
 2008-08-31  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* DLD-FUNCTIONS/fltk_backend.cc (fltk_backend::redraw_figure,
diff --git a/src/symtab.cc b/src/symtab.cc
--- a/src/symtab.cc
+++ b/src/symtab.cc
@@ -188,21 +188,22 @@ out_of_date_check_internal (octave_value
 			  || (nm_len > 2 && nm.substr (nm_len-4) == ".m")))
 		    file = nm;
 		  else
 		    {
 		      // We don't want to make this an absolute name,
 		      // because load_fcn_file looks at the name to
 		      // decide whether it came from a relative lookup.
 
-		      if (dispatch_type.empty ())
-			file = load_path::find_fcn (nm, dir_name);
-		      else
+		      if (! dispatch_type.empty ())
 			file = load_path::find_method (nm, dispatch_type,
 						       dir_name);
+
+		      if (file.empty ())
+			file = load_path::find_fcn (nm, dir_name);
 		    }
 
 		  if (file.empty ())
 		    {
 		      // Can't see this function from current
 		      // directory, so we should clear it.
 
 		      function = octave_value ();
